datafusion/core/src/datasource/physical_plan/arrow_file.rs

```diff
@@ -223,6 +226,65 @@ impl ExecutionPlan for ArrowExec {
     }
 }
 
+/// Arrow configuration struct that is given to DataSourceExec
+/// Does not hold anything special, since [`FileScanConfig`] is sufficient for arrow
+#[derive(Clone, Default)]
+pub struct ArrowConfig {
```

Comment:
Since this implements `FileSource` I would personally suggest a name like `ArrowSource` (and `ParquetSource`, `AvroSource`, etc) for consistency

In my mind this is not really "config" it is the runtime state needed to read arrow files. But I think I could be conviced that Config is ok too.